src/lib.rs

```diff
@@ -75,12 +77,41 @@ impl Order {
         }
     }
 
+    /// Generate a Certificate Signing Request for the given names and request finalization
+    ///
+    /// Uses the rcgen crate to generate a Certificate Signing Request (CSR) for the given
+    /// Subject Alternative Names, then calls [`Order::finalize()`] with it. Returns the generated
+    /// private key, serialized as PEM.
+    ///
+    /// After this succeeds, call [`Order::certificate()`] to retrieve the certificate chain once
+    /// the order is in the appropriate state.
+    #[cfg(feature = "rcgen")]
+    pub async fn finalize(&mut self) -> Result<String, Error> {
```

Comment:
I wonder if we should collapse this down into one operation that returns both the private key & the certificate in one-go. I'm having a hard time thinking of a reason why you'd want to get just the private key and separately `await` the issued certificate (and the `finalize_csr()` route would let you if needed).

In both `provision.rs` and `pebble.rs` we always immediately fetch the cert after the finalize step so that would reduce a bit more boilerplate IMO.

Reply:
Makes sense -- but would like to leave that for a future PR.